### DISCUSSION TITLE
apprarence personnage en /profile et système de compilation serveur

### DISCUSSION THREAD

**Discussion by Naeki-D** [Crownicles/Crownicles#1290] [Nouvelles idées]
Bonsoir, aillant posé la question à pas mal de personnes qui trouvent l'idée original et qui seraient pour, j'aimerai soumettre içi l'idée de pouvoir créer notre personnage.

L'idée serait de choisir une base homme ou femme, quelques coupes de cheveux, yeux, bouche, couleur de peau, le reste s'appliquera (l'équipement) une fois celui ci trouvé et si l'on souhaite l'équiper tout simplement.

Côté serveur là on en demandera plus car il devra avoir en base de données toute les ressources pour compiler une image avec tout les éléments que chaque personne possède, à la demande (l'idée serait de mettre un timer slow mode pour alléger la charge du serveur.

voici quelques exemples de ceux à quoi cela ressemble entre les ressources uniquement et le résultat final :
<img width="760" alt="Capture d’écran 2022-07-24 185330" src="https://user-images.githubusercontent.com/109109099/180657963-84408fde-bc33-4de7-8ddd-0d3ba76ed910.png">

![rWqVxiyijTeXFK9nrKPmXfpr5oQ](https://user-images.githubusercontent.com/109109099/180657970-ce826810-ee32-430f-86c2-0c152e78fabf.png)

au final qui devrait s'occuper de la création de tout ceci ? je me propose bien évidement volontaire, si d'autres graphistes souhaitent m'aider, j'en serais heureux.

**Comment by Pietagorh** [comment]
Ghosts avait dit qu'avec des svg ça devrait pas être trop demandant en performance j'crois
@BastLast tu confirmes ?

**Reply by Arkitu** [reply]
Je précise on peut pas afficher des svg dans discord il me semble

**Comment by Pietagorh** [comment]
Dans le slot d'image en haut à droite dans les embeds dans le profil ce serait stylé j'trouve

**Comment by Voisin1234** [comment]
Ça serait vraiment génial

**Comment by Pietagorh** [comment]
Peut-être faire une petite demo (un croquis quoi) de ton style ? Avec les objets de ton inventaire par exemple

**Comment by BastLast** [comment]
Hello, jai plusieurs soucis avec ce système bien qu'il puisse être stylé.

- la gestion d'image custom a grande échelle est globalement ÉNORMÉMENT plus demandant en perf que du texte et, bien qu'on puisse opti ( svg etc tout ce que vous voulez) ca prendra du temps à coder et ca restera plus lourd que mtn. 

- les graphistes sont durs a trouver et encore plus quand on veut du taf bénévole. Si on fait ca il faut des contacts réguliers et sur le long terme. 

- le fait de ne pas avoir dimage est un des points centraux de draftbot, j'aime beaucoup cette limite et la créativité que ca nous force à avoir tout comme l'imagination des joueurs. 

- avoir cette fonctionnalité peut limiter beaucoup la rédaction en rendant certains events illogique, le fait davoir un personnage communna tous les joueurs plutôt que des trucs perso simplifie la rédaction qui est déjà une tache complexe 

- cette fonctionnalité risque detre compliquée a dev et je pense que lon n'a pas les ressources de dev pour la réaliser.

D'un coup perso je suis vraiment pas chaud pour cette idée meme si je comprends pourquoi elle peut vous plaire.

**Reply by Pietagorh** [reply]
Bah après si on enlève la personnalisation du perso, juste l'affichage de l'inventaire sur un perso que tout le monde a ?
Le plus dur se ferait niveau graphisme, pas dev
Après du coup oui on l'avait déjà évoqué la dernière fois mais faut plusieurs graphistes chauds pour le faire
Et après niveau le gros du travail ce serait juste ajouter les liens des images aux json
Ou alors mettre les images dans resources/images et avec le même format arrangement de fichiers que pour les équipements (faut bien qu'il serve ce dossier xD)

**Reply by Pietagorh** [reply]
(Le temps à coder j'en ai rien à faire tant qu'j'm'amuse x))
Le seul problème du coup c'est les performances quoi
Mais si on a un système pour les persos des joueurs et une charte graphique, on pourra aussi donner des apparences aux personnages créés (genre Gruffo), et ça ce serait tout aussi stylé

Y'a vraiment **aucun** moyen de l'opti pour que ce soit pas trop lourd ? Même en envoyant le message -> edit quand l'image est prête ?

**Reply by BastLast** [reply]
ce n'est pas un problème de disponibilité pr les gens, mais de temps de traitement pr le serv, 
Yaurait moyen de faire ça de manière opti mais ça prendrait je pense beaucoup de temps à faire et ça ne me parait pas trivial (je suis pas sur que le jeu en vaillela chandelle).

Ya vraiment pas mal de taf de graphisme et meme si on en trouve un volontaire pr faire ça mtn dès qu'on va vouloir y toucher va falloir en trouver un autre si il est pas dispo perso c'est le genre de trucs ou je préfèrerai qu'on reste sur du simple.

Après si vraiment il y a besoin d'illu des persos principaux, whynot les réaliser, mais je suis assez peu convaincu de l'utilité de les intégrer au jeu

**Comment by ghost** [comment]
+ quand je parlais de svf, j'avais en tête tous les jeux web du moment et notamment skribb.io ou tous les personnages sont des svg ... je sais que c'est possible mais dans l'ensemble, pour ses jeux, ca represente un élément central du gameplay : pas de visuel, pas de jeu ... draftbot, par son concept de base d'aventure purement textuelle à su pendant deux ans ? se passer de ca en créeant tout un écosystème qui fonctionne plutot bien. le but principal étant d'améliorer le bot sans le complexifier inutilement, cette feature risque de boulverser toutes les habitudes visuelles des joueurs pour une plus value assez mince : bien que ce soit stylé, je penses que tout le monde vas se lasser assez vite de ses illustrations et cela ne rendra pas hommage au travail des artistes... je suis d'accord avec draft sur le fait que commencer cette feature nous engage à travailler de facon permanente avec un ou des artistes. On peut le voir avec l'équipe de dev qui à changé plusieurs fois au grè des évenements, je penses que ce problème vas se rencontrer aussi ici avec que chaque artiste à un style plus ou moins unique donc même si on trouve un autre artiste derriere, la consistance ne sera pas assurée et c'est un élement essentiel... 
cependant, je vois que certaines personnes sont pretes à investir du temps en graphisme pour le projet et il existe bien une solution intermédiaire : les émojis personnalisés ! on commence à faire le tour des émotes de base et on attends impatiament l'arrivée de certains émotes comme l'autruche ... proposer des créations sous forme d'émote en fonctions de l'équipe de rédaction de temps en temps selon les disponibilités et les besoins pourrait être un compromis acceptable selon moi
Pour les illustrations du lore du jeu (personnages, lieux), elles peuvent servir à illustrer le wiki et à le rendre plus interessant !

PS : désolé pour ce pavé
